frontend/src/components/BatchTransfer.tsx

```diff
@@ -20,108 +20,110 @@ import { userAccountState } from '../store';
 
+const isValidAddress = (address: string): boolean => {
+  if(!address) {
+    return false;
+  }
+  return !!address.match(/^0x[0-9a-f]{16}$/)
+}
+
 const BatchTransfer = () => {
   const {
     handleSubmit,
     formState: { errors, isSubmitting },
   } = useForm();
   const [userAccount, setUserAccount] = useRecoilState(userAccountState);
   const [currency, setCurrency] = useState(FLOWCurrency);
-  const [recipientTemplate, setRecipientTemplate] = useState(''); // text area
+  const [txTemplate, setTxTemplate] = useState(''); // text area
+  const [outputs, setOutputs] = useState<Output[]>([])
   
-  const [toAddresses, setToAddresses] = useState<string[]>([]);
-  const [amounts, setAmounts] = useState<string[]>([]);
   const [totalAmount, setTotalAmount] = useState<BigNumber>(new BigNumber(0.0));
   
   const [remaining, setRemaining] = useState<BigNumber>(new BigNumber(0.0));
   const [txHash, setTxHash] = useState('');
   const [txStatus, setTxStatus] = useState(-1)
   
   const [errorText, setErrorText] = useState('');
+  const [validationErrors, setValidationErrors] = useState<ValidationError[]>([]);
   const [checkDone, setCheckDone] = useState(false);
 
   const resetConfirm = () => {
-    setToAddresses([]);
-    setAmounts([]);
+    setOutputs([]);
     setTotalAmount(new BigNumber(0.0));
     setRemaining(new BigNumber(userAccount?.balance[currency.symbol] || 0));
   };
 
   const loadToAddressesAndAmounts = (recipientsAndAmountsStr: string) => {
-    const toAddresses: string[] = [];
-    const amounts: string[] = [];
+    const outputs: Output[] = [];
     recipientsAndAmountsStr
       .split(/\r\n|\n/)
       .filter((v) => !!v)
       .map((recipientsAndAmount) => {
-        const [toAddress, amount] = recipientsAndAmount
+        const [toAddress, amountStr] = recipientsAndAmount
           .split(/[ ,]+/)
           .filter((v) => !!v);
-        toAddresses.push(toAddress);
-        amounts.push(
-          Number(amount || 0)
-            .toFixed(8)
-            .toString()
-        );
-        return [toAddress, Number(amount).toFixed(8).toString()];
+        
+        const amount = Number(amountStr || 0)
+        const output: Output = {
+          address: toAddress,
+          amount: amount,
+          amountStr: amount.toFixed(8).toString(),
+        };
+        outputs.push(output);
       });
-    setToAddresses(toAddresses);
-    setAmounts(amounts);
+    setOutputs(outputs);
 
-    const totalAmount = amounts.reduce(
-      (sum, amount) => sum.plus(new BigNumber(Number(amount) || 0)),
+    const totalAmount = outputs.map(x => x.amount).reduce(
+      (sum, amount) => sum.plus(new BigNumber(amount)),
       new BigNumber(0.0)
     );
     setTotalAmount(totalAmount);
+
     const remaining = new BigNumber(
       userAccount?.balance[currency.symbol] || 0
     ).minus(totalAmount);
     setRemaining(remaining);
 
+    const addressErrors: ValidationError[] = []
+    for(let i=0; i<outputs.length;i++) {
```

Comment:
Just a comment.
Perhaps we should introduce a formatter with `.prettier` file.

Reply:
did `prettier . --write`
https://github.com/flow-japan/batch-token-transfer-tool/pull/3/commits/846e070782fe0cbc0864b241d7d2f5fc118498c1